lib/fc.js

```diff
@@ -428,48 +525,99 @@ async function nasAutomationConfigurationIfNecessary({
       if (definition.isNasAutoConfig(nasConfig)) {
         const yes = await promptForConfirmContinue(`You have already configured 'NasConfig: Auto’. We want to use this configuration to store your function dependencies.`);
         if (yes) {
-          await processNasAutomationConfiguration({ tpl, tplPath, baseDir, runtime, codeUri, nasConfig, vpcConfig,
-            functionProp,
+          await backupTemplateFile(tplPath); // backup tpl
+          await processNasAutomationConfiguration({ tpl, tplPath, baseDir, runtime, codeUri,
             nasServiceName,
             nasFunctionName
           });
+
           stop = true;
-        } else { throw new Error(red(`\nIf 'NasConfig: Auto' is configured, only the configuration store function dependency is currently supported.`)); }
+        } else {
+          throw new Error(red(`\nIf 'NasConfig: Auto' is configured, only the configuration store function dependency is currently supported.`));
+        }
       } else if (!_.isEmpty(vpcConfig) && _.isEmpty(nasConfig)) {
 
         throw new Error(red(`\nFun has detected that you only have VPC configuration. This scenario is not supported at this time. You also need to manually configure the NAS service. You can refer to: https://github.com/alibaba/funcraft/blob/master/docs/specs/2018-04-03-zh-cn.md#nas-%E9%85%8D%E7%BD%AE%E5%AF%B9%E8%B1%A1 and https://nas.console.aliyun.com/`));
       } else if (!_.isEmpty(vpcConfig) && !_.isEmpty(nasConfig)) {
         if (onlyOneNas) {
           const yes = await promptForConfirmContinue(`We have detected that you already have a NAS configuration. Do you directly use this NAS storage function dependencies.`);
           if (yes) {
-            await processNasAutomationConfiguration({ tpl, tplPath, baseDir, runtime, codeUri, nasConfig, functionProp,
+            await backupTemplateFile(tplPath);
+
+            await processNasAutomationConfiguration({ tpl, tplPath, baseDir, runtime, codeUri,
               nasServiceName,
               nasFunctionName
             });
-            stop = true;
-          } else { throw new Error(red('\nYou need to handle function dependencies manually to reduce the size of the package. It is recommended that you use OSS or NAS.')); }
-
+          } else {
+            throw new Error(red('\nYou need to handle function dependencies manually to reduce the size of the package. It is recommended that you use OSS or NAS.'));
```

Comment:
If your yml has been already configured with 'NasConfig', fun only supports to use this 'NasConfig' to process your function dependencies. Otherwise you need to handle the dependencies by yourself.

Refer to https://yq.aliyun.com/articles/712700 for more help.

Reply:
done